pystiche/misc.py

```diff
@@ -280,16 +280,13 @@ def reduce(x: torch.Tensor, reduction: str) -> torch.Tensor:
 
 
 @contextlib.contextmanager
-def suppress_warnings(*categories: Type[Warning]) -> Iterator[None]:
+def suppress_deprecation_warnings(*categories: Type[Warning]) -> Iterator[None]:
     if not categories:
         categories = (UserWarning,)
-    old_filters = set(warnings.filters)  # type: ignore[attr-defined]
     for category in categories:
-        warnings.filterwarnings("ignore", category=category)
-    new_filters = set(warnings.filters) - old_filters  # type: ignore[attr-defined]
-    try:
-        yield
-    finally:
-        for filter in new_filters:
-            warnings.filters.remove(filter)  # type: ignore[attr-defined]
-        warnings._filters_mutated()  # type: ignore[attr-defined]
+        warnings.filterwarnings(
+            "ignore",
+            category=category,
+            message=".* is deprecated since pystiche==.* and will be removed in a future release.",
```

Comment:
I'm a little anxious about matching the message. Can we maybe use the [`module` filter](https://docs.python.org/3/library/warnings.html#describing-warning-filters) to check if a warning originated in `pystiche`?

Reply:
How about `module=.*pystiche`? (https://stackoverflow.com/a/53706908) - I've used this in the recent commit, it works well.

Reply:
See https://github.com/pystiche/pystiche/pull/566#pullrequestreview-763695045.